tests/unit/test_circuitconfig.py

```diff
@@ -0,0 +1,110 @@
+"""
+Unit tests for the CircuitConfig class (internal data structure),
+and the parsing from a circuit info
+which is usually translated from a SONATA config file done in sonata_config.py
+"""
+
+import pytest
+
+from ..conftest import RINGTEST_DIR
+from neurodamus.core.configuration import ConfigurationError, make_circuit_config
+from neurodamus.io.sonata_config import SonataConfig
+
+
+@pytest.mark.parametrize(
+    "create_tmp_simulation_config_file",
+    [
+        {
+            "simconfig_fixture": "ringtest_baseconfig",
+        }
+    ],
+    indirect=True,
+)
+def test_ringtest_circuitconf(create_tmp_simulation_config_file):
+    """
+    test the usual flow: read a sonata config file and convert it to CircuitConfig
+    """
+    config_parser = SonataConfig(create_tmp_simulation_config_file)
+    circuit_dict = config_parser.Circuit.get("RingA")
+    circuit_conf = make_circuit_config(circuit_dict)
+    assert circuit_conf.as_dict() == {
+        "CellLibraryFile": str(RINGTEST_DIR / "nodes_A.h5"),
+        "CircuitTarget": "RingA:Mosaic",
+        "Engine": "METype",
+        "METypePath": str(RINGTEST_DIR / "hoc"),
+        "MorphologyPath": str(RINGTEST_DIR / "morphologies/asc"),
+        "MorphologyType": "asc",
+        "PopulationType": "biophysical",
+        "nrnPath": str(RINGTEST_DIR / "local_edges_A.h5:RingA__RingA__chemical"),
+    }
+
+
+def test_empty_circuit():
```

Comment:
I think I'm behind on something, but why is an empty circuit needed?
Maybe in another commit we can make a type to designate an `DummyCircuitConfig`, so it's not dependent on checking that certain properties that normally are `str` are `bool`, or that the sentinel value "<NONE>" is used?

Reply:
In the `main` branch, the `SimConfig._base_circuit` is an empty circuit reserved for BlueConfig circuit. In the SONATA words, this empty circuit is used to instantiate an empty cell distributor to hold a fake cell in CoreNEURON case in certain ranks (when no. of ranks > no. of cells)
https://github.com/openbraininstitute/neurodamus/blob/main/neurodamus/node.py#L536-L540
In this PR, I still need the same creation when a fake cell is needed. 
As we have the `make_circuit_config` function to create `CircuitConfig` obj from a dict, why not to use it to create a dummy CircuitConfig too?